### PR TITLE
Modifying front-end process to use Sauce Tunnel

### PR DESCRIPTION
**Pull request by sebworks**
Modifying front-end process to use Sauce Tunnel

## Changes

- Modified `config/environment.js ` to add `SAUCE_TUNNEL` and default value. 
- Modified `gulp/tasks/test.js` to create a Sauce tunnel workflow and handle command line params. 
- Modified `npm-shrinkwrap.json` to remove unnecessary dependencies. 
- Modified `package.json` to add `sauce-tunnel`.
- Modified `test/browser_tests/conf.js` to add a `_useSauceLabs` method. This also changes the way the default sauce flag works. You only run on sauce when passing the flag, which is more aligned with the expected testing behavior. 
- Modified `test/browser_tests/default-suites.js` to remove IE8 and switch to Windows 7. 
- Modified `tox.ini` to use the appropriate variables. 

## Testing

1. Run `npm install`.
2. Run `gulp build`. 
3. Run `gulp test:acceptance --fast` and verify that the tests should run as normal.
4. Set your Sauce credentials in the `.env` file.
5. Run `gulp test:acceptance --fast --sauce`  and observe the log as the tunnel is created and the tests start.  Login to sauce labs and verify that the test results match those viewed locally.

## Screenshots

- 
screen shot 2017-06-15 at 3 28 08 pm

## Notes

- The Sauce Tunnel package contains binaries which are 40MB. This is larger than most of our other NPM packages. It might make sense to switch to a different package which lazy loads the binaries on first usage. 

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -18,11 +18,13 @@ var envvars = {
   SAUCE_USERNAME:          process.env.SAUCE_USERNAME,
   SAUCE_ACCESS_KEY:        process.env.SAUCE_ACCESS_KEY,
   ACHECKER_ID:             process.env.ACHECKER_ID,
-  HEADLESS_CHROME_BINARY:  process.env.HEADLESS_CHROME_BINARY
-
+  HEADLESS_CHROME_BINARY:  process.env.HEADLESS_CHROME_BINARY,
+  SAUCE_TUNNEL:            process.env.SAUCE_TUNNEL ||
+                           new Date().getTime()
```
Comment (sebworks) [inline config/environment.js]: yeah, that is a good point.

**Comment by sebworks** [comment]
> Works! My tests are timing out, but they're sending to Sauce.

@anselmbradford, which ones are timing out?

**Comment by anselmbradford** [comment]
screen shot 2017-06-19 at 9 06 19 am

**Comment by sebworks** [comment]
@anselmbradford, please remove the GA code in base.html and retest.

**Comment by anselmbradford** [comment]
@sebworks I get the same thing.

**Comment by sebworks** [comment]
I'm placing this on hold until I fork the repo. The owner of the sauce-tunnel repo isn't upgrading the binaries.

**Comment by sebworks** [comment]
@anselmbradford, some dude had already forked it. I switched to his module.

**Pull request merged**